### ISSUE TITLE
custom metric MAE and RMSE are the same

### ISSUE THREAD

**Issue by germayneng** [keras-team/keras#10706]
This is a reproducible example: 

consider this 

```
from keras.models import Sequential
from keras.layers import Dense, Activation
import numpy as np








import keras.backend as K

def rmse (y_true, y_pred):
    return K.sqrt(K.mean(K.square(y_pred -y_true), axis=-1))




model = Sequential([
    Dense(32, input_shape=(50,)),
    Activation('relu'),
    Dense(1),
])

model.compile(optimizer='adam', loss='mse', metrics=['mae',rmse])





data = np.random.random((1000, 50))
labels = np.random.randint(2, size=(1000, 1))

model.fit(data, labels, epochs=5, batch_size=64)
```

Both RMSE and MAE are the same. This happens when i placed them together. Is there a bug?

the output is:

![image](https://user-images.githubusercontent.com/22788747/42806109-39fc16be-89e0-11e8-80d8-5e73ba8b2541.png)

**Comment by tRosenflanz** [comment]
Rmse calculation is wrong. Try to get rid of `axis=-1` - I had problems with RMSE before and removing the axis argument worked for me

**Comment by germayneng** [comment]
thanks man. Without the axis = -1, it seems to be correct. The axis -1 is based on the original metrics. May i ask why was the axis -1 needed in the first place? (for the Original metrics defined in tf)

**Comment by tRosenflanz** [comment]
I have no idea in this case to be honest - if nothing else it should be axis=0. Otherwise it was doing square, mean of each number (i.e. nothing) and square root of the square (i.e. absolute value of original calculation) so it was just returning mae

**Comment by germayneng** [comment]
yes thinking about it, the mean should be computed along axis = 0, otherwise, it will simply return the original vector.

but in the case of axis = -1, wouldnt there be an error since we still have a vector even after the sqrt?

**Comment by tRosenflanz** [comment]
It is possible that somewhere in Keras code it calls the mean on the vector
since most losses have mean as the last operation (mse, Mae, logloss, etc).
I will investigate tomorrow

**Comment by tRosenflanz** [comment]
Update:
At least in losses we have this line that explicitly calls K.mean on the loss 
https://github.com/keras-team/keras/blob/f86bc57d621fb0325e928271b246d197611b0e65/keras/engine/training_utils.py#L422

**Comment by tRosenflanz** [comment]
Okay, this explains the behaviour pretty well at this point - although this is not documented explicitly from what I can tell- the losses and metrics are computed based on the mean implicitly. Can you please close the issue ?

**Comment by uwu-ai** [comment]
@tRosenflanz Hey so i am running into the same issue, I have my RMSE updated to 

```def rmse(y_true, y_pred):
    return K.sqrt(K.mean(K.square(y_pred - y_true)))```

and yet the RMSE and MAE values are the same. the RMSE is also not equal to the root of the MSE.

**Comment by tRosenflanz** [comment]
@uwu-ai what version of Keras/tf and what is the shape of your output?
Have you tried passing different axis arguments into the mean?

**Comment by uwu-ai-ta** [comment]
@tRosenflanz hey (using a diff account cause had login issues) 
keras == 2.3.1
tensorflow == 2.2.0

Should mention this is a regression task, the output shape should be (1, 1). 
Yes i tried passing axis=[-1, -2] and axis=0, same results.

**Comment by tRosenflanz** [comment]
Output shape is 1,1 or Batch_size,1,1?

**Comment by uwu-ai-ta** [comment]
Sorry model.summary() shows output shape from last layer is (None, 1).

**Comment by tRosenflanz** [comment]
Does your generator output sample weights by any chance?

**Comment by uwu-ai-ta** [comment]
@tRosenflanz  Yes it does.

**Comment by tRosenflanz** [comment]
Can you disable that and see what loss you get? This way it will be easier to understand how rmse and mse actually correlate

**Comment by uwu-ai-ta** [comment]
@tRosenflanz So i removed the sample_weights and got the following result (showing only 1 epoch but the rest four follow the same)
```
Epoch 1/1000
45/45 [==============================] - 42s 928ms/step - loss: 2.6088 - mae: 0.2379 - rmse: 0.2379 - val_loss: 2.5272 - val_mae: 0.2618 - val_rmse: 0.2618
```
